src/backend/distributed/commands/function.c

```diff
@@ -192,6 +198,45 @@ create_distributed_function(PG_FUNCTION_ARGS)
 
 	ObjectAddressSet(functionAddress, ProcedureRelationId, funcOid);
 
+	DistObjectCacheEntry *cacheEntry =
+		LookupDistObjectCacheEntry(ProcedureRelationId, funcOid, InvalidOid);
+	if (cacheEntry != NULL && IsObjectDistributed(&functionAddress))
+	{
+		/*
+		 * If the colocationId, forceDelegation and distributionArgIndex fields of a
+		 * pg_dist_object entry of a distributed function are all set to zero, it means
+		 * that function is either automatically distributed by ddl propagation, without
+		 * calling create_distributed_function. Or, it could be distributed via
+		 * create_distributed_function, but with no parameters.
+		 *
+		 * For these cases, calling create_distributed_function for that function,
+		 * without parameters would be idempotent. Hence we can simply early return here,
+		 * by providing a notice message to the user.
+		 */
+		bool idempotentDistribution = /* are pg_dist_object fields set to zero? */
+									  cacheEntry->colocationId == 0 &&
+									  cacheEntry->forceDelegation == 0 &&
+									  cacheEntry->distributionArgIndex == 0 &&
+
+		                              /* called without any parameters? */
+									  distributionArgumentName == NULL &&
+									  colocateWithTableNameDefault &&
+									  forceDelegationAddress == NULL;
+
+		if (idempotentDistribution)
+		{
+			ereport(NOTICE, (errmsg("function is already distributed"),
+							 errdetail("Called create_distributed_function without "
+									   "parameters"),
+							 errhint("You can use create_distributed_function without "
```

Comment:
We can update it to sth like "If you want update distribution_argument_index, colocationid or force_delegation parameters for already distributed function, please provide parameters"